src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -40,167 +29,142 @@ const distanceToFurthestCorner = (x: number, y: number, rect: ClientRect) => {
  * @docs-private
  */
 export class RippleRenderer {
-  private _backgroundDiv: HTMLElement;
-  private _rippleElement: HTMLElement;
+
+  /** Element where the ripples are being added to. */
+  private _targetElement: HTMLElement;
+
+  /** Element which triggers the ripple elements on mouse events. */
   private _triggerElement: HTMLElement;
-  _opacity: string;
-
-  constructor(_elementRef: ElementRef,
-              private _eventHandlers: Map<string, (e: Event) => void>,
-              private _ngZone: NgZone) {
-    this._rippleElement = _elementRef.nativeElement;
-    // The background div is created in createBackgroundIfNeeded when the ripple becomes enabled.
-    // This avoids creating unneeded divs when the ripple is always disabled.
-    this._backgroundDiv = null;
-  }
 
-  /** Creates the div for the ripple background, if it doesn't already exist. */
-  createBackgroundIfNeeded() {
-    if (!this._backgroundDiv) {
-      this._backgroundDiv = document.createElement('div');
-      this._backgroundDiv.classList.add('md-ripple-background');
-      this._rippleElement.appendChild(this._backgroundDiv);
-    }
+  /** Whether the mouse is currently down or not. */
+  private _isMousedown: boolean = false;
+
+  /** Currently showing ripples that will be closed on mouseup. */
+  private _showingRipples: HTMLElement[] = [];
+
+  /** Events to be registered on the trigger element. */
+  private _triggerEvents = new Map<string, any>();
+
+  /** Ripple config for all ripples created by events. */
+  rippleConfig: RippleConfig = {};
+
+  /** Whether mouse ripples should be created or not. */
+  rippleDisabled: boolean = false;
+
+  constructor(_elementRef: ElementRef, private _ngZone: NgZone, private _ruler: ViewportRuler) {
+    this._targetElement = _elementRef.nativeElement;
+
+    // Specify events which need to be registered on the trigger.
+    this._triggerEvents.set('mousedown', this.onMousedown.bind(this));
+    this._triggerEvents.set('mouseup', this.onMouseup.bind(this));
+    this._triggerEvents.set('mouseleave', this.onMouseLeave.bind(this));
```

Comment:
Rather than using a `Map`, how about an array of tuples? Better for iteration
```ts
let triggerEvents: [string, (MouseEvent) => void][] = [
    ['mousedown', e => this.onMouseDown(e)],
    // ...
];
```
(generally prefer arrow functions to `.bind()`)